File: greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


int main(void) 
{

    float amt;
    
    do
    {
    printf("O hai! How much change is owed? $ ");
    amt = get_float();
    }
    while (amt < 0);
    
    amt = round(amt * 100);
    
    int cents = amt;
    int quarter = cents / 25;
    int quart_mod = cents % 25;
    int dime = quart_mod / 10;
    int dime_mod = quart_mod % 10;
    int nickel = dime_mod / 5;
    int nickel_mod = dime_mod % 5;
    int penny = nickel_mod / 1;
    
    printf("%i\n", quarter + dime + nickel + penny);

}

File: greedy.py
import cs50

#Determine change by dividing amount with largest coin and repeating for remainder using all smaller coins
def main():
    amt = get_amt()
    quarter = round(amt // 25)
    amt = round(amt % 25)
    dime = round(amt // 10)
    amt = round(amt % 10)
    nickel = round(amt // 5)
    amt = round(amt % 5)
    penny = round(amt // 1)
    #Print total number of coins required for change
    print("{}".format(quarter + dime + nickel + penny))
    
#Prompt user for amount    
def get_amt():
    #Repeat until non-negative value provided
    while True:
        print("O hai! How much change is owed? $", end=" ")
        amt = cs50.get_float()
        if amt >= 0:
            break
    #Multiply amount by 100 to convert dolalrs to cents
    amt = amt * 100
    #Return amount to main function
    return amt

#Call main function    
if __name__ == "__main__":
    main()